spec/datadog/tracing/workers_integration_spec.rb

```diff
@@ -243,13 +243,13 @@ def wait_for_flush(num = 1, period = 0.1)
     end
 
     context 'which overruns the timeout' do
-      let(:task) { proc { sleep(interval) } }
+      let(:task) { proc { sleep(10) } }
```

Comment:
Because the now has a fixed lenience of 5 seconds (`be_within(5)`), I changed the background thread sleep time to match that fixed constraint: the job will try to sleep for 10 seconds, but should shut down earlier than that to pass.

Reply:
I wonder if it would make sense to even convert this to a regular `sleep()` with no timeout, since this test is explicitly testing the interruption, and thus the  `sleep` is never expected to finish by itself.

Reply:
I wary of infinite sleeps because the test might get hung up forever in the future for some reason.
I set this timeout much larger than it should be, but still not too large to hand CI.

Reply:
Hmmm, fair enough, although I somewhat worry about the potential for "accidentally passes for the wrong reason".